V2 upgrade

wkdewey opened the pull request.

wkdewey committed:
upgrade ruby to 3.1.7 and datura to 0.1.7

wkdewey committed:
upgrade to api-v2

wkdewey committed:
update fields for api-v2

wkdewey committed:
fix error handling when posting/webscraping

wkdewey committed:
make some additional adjustments, fix bugs

Commit:
update public.yml and remove private endpoints

Commit:
remove reference to xslt file no longer in Datura

techgique commented on Gemfile:
```diff
@@ -1,4 +1,5 @@
-gem "datura", git: "https://github.com/CDRH/datura.git", tag: "v0.2.0-beta"
+gem "datura", git: "https://github.com/CDRH/datura.git", tag: "v0.1.7"
 
+gem "byebug"
```
Byebug should be dropped from this specific repo's Gemfile as it is now a dependency of Datura itself and will always be present for collection repos.

wkdewey replied:
removed

techgique commented on scripts/overrides/tei_to_es.rb:
```diff
@@ -148,28 +148,26 @@ def has_source
 
   def spatial
-    get_elements(@xpaths["spatial"]).map do |ele|
+    get_elements(@xpaths["spatial"]).select{}.map do |ele|
```
Does this `.select{}` here do anything when there are no arguments to it? Doesn't that just return the same array it was called on? Guessing this can be removed.

wkdewey replied:
I _think_ what I was trying to do ended up being covered in the code below. But see https://github.com/CDRH/family_letters/issues/181 for a somewhat related issue.

techgique replied:
Fix confirmed after re-posting data. Thanks!

wkdewey committed:
remove redundant byebug dependency

wkdewey committed:
add all location values to the API

fixes https://github.com/CDRH/family_letters/issues/181

techgique committed:
Update required bundler version

Newer server was failing to post until `bundle update --bundler`
was run, which updates Gemfile.lock. Committing that change.
